### ISSUE TITLE
Enhancement: Handling of diagram size

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#11117]
### Description

The users of our BRSL library often simulate large hydraulic circuits. During this they often use hydraulic circuit diagrams as a starting point and as an orientation. So its sometimes easier to not create some sub-models to have a similar look of the circuit diagram and the simulation model.
However, changing the size of the diagram layer (or more specifically its white rectangle) can only be done via the text layer in OMEdit.

### Steps to Reproduce

1. Consider the following modelica model.
```modelica
model DiagramSize
  Modelica.Blocks.Sources.Constant const annotation (Placement(transformation(extent={{-20,180},{0,200}})));
  annotation (uses(Modelica(version="4.0.0")));
end DiagramSize;
```
2. In e.g. Dymola the diagram layer looks as shown in the first screenshot. The green points appear when I click on the boundary of the diagram extent and let the user dynamically change its size. The user is also able to zoom to the constant block despite it being outside the white rectangle (second screenshot).
3. When opening the model in OMEdit there is no equivalent to the "green points". I am also not able to recreate the second picture because when zooming into the model the scrollbar only reaches to the top of the white area (third screenshot). The Diagram size can only be changed by adding a new diagram annotation from scratch in the text layer. 

### Expected Behavior

1. There is a way to change the size of the diagram layer by "dragging" the white rectangle's boundaries in the diagram view. Consequently, there should be no need to change the diagram annotation's extent argument in the text layer, which is not even present when opening a new model. 
4. When there are instances outside of the white triangle, the user is able to zoom to them. This point looses most of its relevance if point 1 is possible.

### Screenshots

My colleague @niklwors will add the screenshots as comments.

### Version and OS

 - OpenModelica Version: v1.22.0-dev-278-ge7b7ec3ee1 (64-bit)
 - OS: Windows 10, 64 bit
 - MSL 4.0.0

### Additional Context

Thanks for considering this 
Aaron Buntrock - Bosch Rexroth

**Comment by niklwors** [comment]
picture 1:
![image001](https://github.com/OpenModelica/OpenModelica/assets/12269776/66b3dadc-dd86-426e-88f3-aaa5ce809405)


picture 2:
![image002](https://github.com/OpenModelica/OpenModelica/assets/12269776/445a251d-4c99-4733-89b6-e0c1f108a6a0)

picture 3:
![image003](https://github.com/OpenModelica/OpenModelica/assets/12269776/d27d72f7-c0b5-4fcf-a861-d593ced2b400)

**Comment by ceraolo** [comment]
In OM there's a way to modify the white page which is much faster and more effective than Dymola's.
Simply put an object un the green part of the screen, where you want, and click on the "Fit to diagram" button

**Comment by abuntrock** [comment]
@ceraolo thanks for the hint. I totally missed that button for this case. This is the solution to my problem.

**Comment by ceraolo** [comment]
I asked for this feature some time ago.
I hope you'll agree that it is much better than Dymola's solution.
It is very useful also to eliminate blank areas from diagram sheets, which is conveninet when we want to see diagrams in the plotting perspective: OM automatically zooms in such a way that the white page is displayed.

@adeas31 I suppose you can close this ticket?

**Comment by abuntrock** [comment]
Yeah it definitley ticks all the boxes I except from such a solution. I think I already closed this issue.
